### PR TITLE
Add validations on metadata form

### PR DESCRIPTION
**Pull request by anukritijha**
### Type of change

- Enhancement / new feature

### Description

This PR adds the following features to metadata form

- Adds dynamic child rows, its key, values and types in metadata json 
- Enables/Disables the value field in case of Array or Object
- Enables/Disables the child row addition button
- Removes the generic typeahead component to patternfly select component
- Adds validations to metadata fields

image

### CODE REVIEW

File: console/console-init/ui/src/components/TypeAheadSelect/TypeAheadSelect.tsx [hunk 1]
```diff
@@ -75,6 +75,7 @@ const TypeAheadSelect: React.FunctionComponent<ITypeAheadSelectProps> = ({
           } else {
             setOptions([
               <SelectOption
+                key={0}
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/components/TypeAheadSelect/TypeAheadSelect.tsx:78]: can we replace the key with dynamic value like index?

File: console/console-init/ui/src/modules/device/dialogs/CreateDevice/MetaDataRow.tsx [hunk 2]
```diff
@@ -11,108 +11,165 @@ import {
...
-            />
-            <Button
-              variant="control"
-              aria-label="Add child on button click"
-              onClick={e => handleAddChildRow(e, "context")}
-            >
-              <PlusIcon />
-            </Button>
+            ></Select>
+            {isChildAdditionEnabled === true ? (
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/modules/device/dialogs/CreateDevice/MetaDataRow.tsx:139]: we can replace ternary operator by && like {isChildAdditionEnabled && (
<Button
                variant="control"
                aria-label="Add child on button click"
                onClick={e => {
                  let parentValue: string = metadataRow.key;
                  handleAddChildRow(e, parentValue);
                }}
              >
                <PlusIcon />
              </Button>
)}

File: console/console-init/ui/src/modules/device/utils/constant.ts [hunk 3]
```diff
@@ -60,4 +60,17 @@ const getInitialFilter = () => {
   return filter;
 };
 
-export { deviceTypeOptions, deviceStatusOptions, getInitialFilter };
+const deviceRegistrationTypeOptions: ISelectOption[] = [
+  { key: "string", label: "String", value: "String" },
+  { key: "number", label: "Number", value: "Number" },
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/modules/device/utils/constant.ts:65]: label should be Numeric

Comment (drcoolsanjeev) [reply console/console-init/ui/src/modules/device/utils/constant.ts:65]: also value will be in lower case

File: console/console-init/ui/src/components/TypeAheadSelect/TypeAheadSelect.tsx (hunk 1 above)
Comment (anukritijha) [reply console/console-init/ui/src/components/TypeAheadSelect/TypeAheadSelect.tsx:78]: There is no dynamic value here, as there is only one element.

File: console/console-init/ui/src/modules/device/dialogs/CreateDevice/MetaDataRow.tsx [hunk 4]
```diff
@@ -11,108 +11,165 @@ import {
...
+    updatedValueMetadata[rowIndex].value = propertyValue;
+    setMetadataList(updatedValueMetadata);
   };
 
   const onPropertySelect = (e: any, selection: SelectOptionObject) => {
     setPropertySelected(selection.toString());
-    setPropertyInput(undefined);
   };
 
   const onPropertyClear = () => {
```
Comment (rkpattnaik780) [inline console/console-init/ui/src/modules/device/dialogs/CreateDevice/MetaDataRow.tsx:110]: Not able to clear the input using the close icon.

File: console/console-init/ui/src/modules/device/dialogs/CreateDevice/MetaData.tsx [hunk 5]
```diff
@@ -4,36 +4,39 @@
...
+      ...metadataList,
+      { key: "", value: [], type: defaultType }
+    ]);
   };
 
   return (
     <>
       <MetaDataHeader sectionName="Default properties" />
-      {formData.map((data, index) => (
+      {metadataList.map((matadataRow, index) => (
```
Comment (rkpattnaik780) [inline console/console-init/ui/src/modules/device/dialogs/CreateDevice/MetaData.tsx:34]: `metadatarow` is not being used anywhere.

**Pull request closed without merging**